### PR TITLE
Add buildkite pipeline

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+steps:
+  - label: "GPU Integeration"
+    plugins:
+      - JuliaCI/julia#v0.5:
+          version: '1.5'
+      - JuliaCI/julia-test#v0.3:
+          test_args: "--thorough"
```
Comment (maleadt) [inline .buildkite/pipeline.yml:7]: that's CUDA.jl-specific

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    branches: "!gh-pages"
+    agents:
+      queue: "juliagpu"
+      cuda: "11.1"
```
Comment (maleadt) [inline .buildkite/pipeline.yml:11]: Do you require this? Pleasae don't specify it if not

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    env:
+      JULIA_CUDA_VERSION: '11.1'
+      JULIA_CUDA_USE_BINARYBUILDER: 'true'
```
Comment (maleadt) [inline .buildkite/pipeline.yml:14]: Same here, please don't specify these env vars unless you really need them.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (DhairyaLGandhi) [reply .buildkite/pipeline.yml]: Done

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (DhairyaLGandhi) [reply .buildkite/pipeline.yml]: removed the extra variables
Thanks!

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+steps:
+  - label: "GPU Integeration"
+    plugins:
+      - JuliaCI/julia#v0.5:
+          version: '1.5'
```
Comment (CarloLucibello) [inline .buildkite/pipeline.yml:5]:
```suggestion
          version: '1'
```
?

**Comment by CarloLucibello** [comment]
Is this going to fix the timeouts we are seeing lately in Zygote (e.g. https://github.com/FluxML/Zygote.jl/pull/841)?

File: .buildkite/pipeline.yml (hunk 4 above)
Comment (maleadt) [reply .buildkite/pipeline.yml:5]: That's not supported by https://github.com/JuliaCI/julia-buildkite-plugin

**Comment by maleadt** [comment]
You don't need Bors with Buildkite, so I assume it'll fix that issue.

**Comment by DhairyaLGandhi** [comment]
Yeah, the resources for the GPU CI are essentially being shifted over

File: .buildkite/pipeline.yml (hunk 4 above)
Comment (johnnychen94) [reply .buildkite/pipeline.yml:5]: This can be supported if using https://github.com/johnnychen94/jill.py

**Pull request merged**